### PR TITLE
Instrument the restart-required flows (closes #1375).

### LINKED ISSUE #1375: Insturment the restart-required flows when using mozAddonManager
Breaking out of #1360.

My suggestion from that thread:

> - Event: user sees CTA
>   - Dimension: is user on homepage or experiment detail page?
>   - Dimension: can user use `mozAddonManager`?
> - Event: user clicks install button
>   - Dimension: is restart required?
> - Event: user shown "must restart" modal.
> - Event: add-on opens homepage after restart
>   - We'd want `utm_`s for this.
> 
> This should let us segment different pathways as they're fed into existing funnels to measure the effect on conversion rates. Plus, we would have absolute numbers to pair with the conversion rates to serve as a basis for speculation of how many users we lost in each step.

CC @ckprice

### PR DESCRIPTION
**Pull request by chuckharmston**
r? @ckprice

### CODE REVIEW

**Comment by chuckharmston** [comment]
Note: the UTMs were added in https://github.com/mozilla/testpilot/pull/1379, where I added the restart-tab-opening.

File: frontend/src/app/lib/addon.js [hunk 1]
```diff
@@ -46,8 +46,9 @@ export function installAddon(store, eventCategory, experimentTitle) {
   };
 
   if (useMozAddonManager) {
-    sendToGA('event', gaEvent);
     mozAddonManagerInstall(downloadUrl).then(() => {
+      gaEvent.eventValue = RESTART_REQUIRED ? 1 : 0;
```
Comment (ckprice) [inline frontend/src/app/lib/addon.js]: `RESTART_NEEDED`?

Comment (chuckharmston) [reply frontend/src/app/lib/addon.js]: ...yeah. I don't know why this worked when I tested it out, I'm going to backtrack to make sure it's doing what's intended.

Comment (chuckharmston) [reply frontend/src/app/lib/addon.js]: At least the tests failed.

**Review by lmorchard — APPROVED** [review]
Code looks :100: to me.

**Comment by pdehaan** [comment]
Needs a rebasin', but not sure if we want to fix the "Insturment" typo in commit message while we're at it. :trollface:

File: frontend/src/app/lib/addon.js [hunk 2]
```diff
@@ -46,8 +46,9 @@ export function installAddon(store, eventCategory, experimentTitle) {
   };
 
   if (useMozAddonManager) {
-    sendToGA('event', gaEvent);
     mozAddonManagerInstall(downloadUrl).then(() => {
+      gaEvent.eventValue = RESTART_NEEDED ? 1 : 0;
```
Comment (pdehaan) [inline frontend/src/app/lib/addon.js:50]: Is this just casting a Boolean to an int? And if so, couldn't you just do `gaEvent.eventValue = Number(RESTART_NEEDED);`?

``` sh
$ node -e console.log(Number(true));  # 1
$ node -e console.log(Number(false)); # 0
```

Comment (chuckharmston) [reply frontend/src/app/lib/addon.js:50]: Yeah, but when you're giving numbers magic values, I'd rather be explicit about what the range of values are. Clear > cute.

**Comment by ckprice** [comment]
I pulled the branch down, and there were no errors, but I wasn't able to fully test the small measurement pieces (eventValue and utm values). I ventured down https://github.com/mozilla/testpilot/pull/1330#issuecomment-245821762 to enable `mozAddonManager`, but ended up doing more harm than good here I think. Can we merge this in, and I'll take an action item to test on `dev` (assuming it's any easier there)?

**Comment by chuckharmston** [comment]
Sure. Merging.

**Pull request merged**